pkg/config/config.go

```diff
@@ -90,3 +87,22 @@ func detectGoVersion() string {
 
 	return "1.17"
 }
+
+// Trims the Go version to keep only M.m.
+// Since Go 1.21 the version inside the go.mod can be a patched version (ex: 1.21.0).
+// The version can also include information which we want to remove (ex: 1.21alpha1)
+// https://go.dev/doc/toolchain#versions
+// This a problem with staticcheck and gocritic.
+func trimGoVersion(v string) string {
+	if v == "" {
+		return ""
+	}
+
+	exp := regexp.MustCompile(`(\d\.\d+)(?:\.\d+|[a-z]+\d)`)
```

Comment:
cosmetic: move it in package level (panic + performance)?

Reply:
it's not useful: it's not in the critical path, and it's just a detail of configuration validation, we don't need high performance here.
The panic will never happen because the expression is a hard-coded string.